### DISCUSSION TITLE
How to use block code and xbox controller to achieve LEGO 42114 and 42131 gear shift operation

### DISCUSSION THREAD

**Discussion by myhewen** [pybricks/support#1546] [Q&A]
How to use block code and xbox controller to achieve LEGO 42114 and 42131 gear shift operation? Can you give me a picture tutorial? Thank you!

**Comment by laurensvalk** [comment]
Do you happen to have the LEGO Train remote? If so, you could try the 42131 Bulldozer with this program?

I just made a test program, but I don't have the `42131` myself to test it :slightly_smiling_face: . Thank you!

[**Download the file here**](https://www.dropbox.com/scl/fi/ku7ze7m5ja3lwpgscz59b/set_42131_bulldozer.py?rlkey=lsqoqlsepqfpi1sy9ue1dk3ai&dl=1)

It's supposed to work like [this video](https://www.youtube.com/watch?v=N7OmW-7u8fA)

Just click the import program button to open it.

![image](https://github.com/pybricks/support/assets/12326241/6bdf174c-4228-45cb-8eb4-bc8ce59f4e5d)




You will get this:

![pybricks_program_2024-03-15T15_44_23](https://github.com/pybricks/support/assets/12326241/7abf6199-97d1-4e30-acc2-97ded297e580)


Looking forward to hearing your results!

**Reply by laurensvalk** [reply]
I will also make an Xbox version for you today or tomorrow.

**Comment by myhewen** [comment]
Thank you!

**Comment by laurensvalk** [comment]
Here is the variant for the Xbox for the `42131` bulldozer. Please let us know how it goes! Thanks!

You can drive with the joysticks. Select a function with A/B/X/Y. Then run the function with the bumpers (RB/LB).

[**Download the file here**](https://www.dropbox.com/scl/fi/qfo5ie6awmf5mbqsscsqw/set_42131_bulldozer_xbox.py?rlkey=rzgu848my1ba5og7hgftc7kzg&dl=1)

![pybricks_program_2024-03-15T15_43_47](https://github.com/pybricks/support/assets/12326241/d2ccf1c8-2c60-47e2-87c6-8f82475867f6)

**Comment by laurensvalk** [comment]
I just launched a new example on the site: https://pybricks.com/project/technic-42131-xbox/

If you want to test it and confirm that it works, that would be awesome. Thank you!
